### ISSUE TITLE
Custom Headers on login

### ISSUE THREAD

**Issue by hegerdes** [lucasheld/uptime-kuma-api#20]
Thank for this API since the official one is still missing.

I would really want to use Uptime-Kuma but we need programmatic access to the API for configuration, so this API is the perfect fit (even with ansible support 👍). But I use custom authentifikation which is handelnd by a reverse proxy, so that I can use oauth and other nice features.

Unfortunately the API only supports Uptime-Kuma own auth mechanism and the is no native way to change it. It would be really nice to allow custom auth or at least set extra header on the initial socketIO connection. 

Right now I solved it by overriding the `connect` method:
```python
class UptimeKumaApiExtended(UptimeKumaApi):

    def connect(self) -> None:
        """
        Connects to Uptime Kuma.

        Called automatically when the UptimeKumaApi instance is created.

        :raises UptimeKumaException: When connection to server failed.
        """
        url = self.url.rstrip("/")
        try:
            self.sio.connect(f'{url}/socket.io/', wait_timeout=self.wait_timeout, headers=headers)
        except:
            raise UptimeKumaException("unable to connect")
```
Is there a option to support this use case naively?

If you are open for collaboration I can provide a example implementation in the near future.

**Comment by lucasheld** [comment]
Sure. Custom headers can be implemented like this: https://github.com/lucasheld/uptime-kuma-api/compare/master...feature/custom-headers

> It would be really nice to allow custom auth or at least set extra header on the initial socketIO connection.

What do you mean by "custom auth or at least set extra header"? What other ways exist to implement the authentification?

The connect method also has a `auth` parameter. Is it useful?
https://python-socketio.readthedocs.io/en/latest/api.html?highlight=auth#socketio.AsyncClient.connect

**Comment by lucasheld** [comment]
Included in [release 0.13.0](https://github.com/lucasheld/uptime-kuma-api/releases/tag/0.13.0).
